drivers/leds/leds-aw2013.c

```diff
@@ -0,0 +1,486 @@
+// SPDX-License-Identifier: GPL-2.0+
+//
+// Driver for Awinic AW2013 3-channel LED driver
+// Based on leds-an30259a.c
+//
+// Datasheet:
+// https://datasheet.lcsc.com/szlcsc/Shanghai-Awinic-Tech-AW2013DNR_C252440.pdf
+
+#include <linux/i2c.h>
+#include <linux/leds.h>
+#include <linux/module.h>
+#include <linux/regulator/consumer.h>
+#include <linux/mutex.h>
+#include <linux/of.h>
+#include <linux/regmap.h>
+
+#define AW2013_MAX_LEDS 3
+
+/* Reset and ID register */
+#define AW2013_RSTR 0x00
+#define AW2013_RSTR_RESET 0x55
+#define AW2013_RSTR_CHIP_ID 0x33
+
+/* Global control register */
+#define AW2013_GCR 0x01
+#define AW2013_GCR_ENABLE BIT(0)
+
+/* LED channel enable register */
+#define AW2013_LCTR 0x30
+#define AW2013_LCTR_LE(x) BIT((x))
+
+/* LED channel control registers */
+#define AW2013_LCFG(x) (0x31 + (x))
+#define AW2013_LCFG_IMAX_MASK (BIT(0) + BIT(1)) // Should be 0-3
```

Comment:
```suggestion
#define AW2013_LCFG_IMAX_MASK (BIT(0) | BIT(1)) // Should be 0-3
```

possibly even `GENMASK(...)` but I always forget how to use it....